logdna/adapter/adapter.go

```diff
@@ -153,47 +141,45 @@ func (adapter *Adapter) Stream(logstream chan *router.Message) {
 // readQueue is a method for reading from queue:
 func (adapter *Adapter) readQueue() {
 
-    buffer := make([]Line, 0)
-    timeout := time.NewTimer(adapter.Limits.FlushInterval)
-    byteSize := 0
+    buffer := make([]Line, 0, adapter.Config.MaxBufferSize)
+    timeout := time.NewTimer(adapter.Config.FlushInterval)
 
     for {
         select {
         case msg := <-adapter.Queue:
-            if uint64(byteSize) >= adapter.Limits.MaxBufferSize {
+            if len(buffer) >= cap(buffer) {
```

Comment:
Not a huge fan of this, I'd rather compare to `adapter.Config.MaxBufferSize` the the capacity.

Reply:
I see the issue now - I compare 2 different things - `MaxBufferSize` is a size in `MB` but the other one is just # of `Line`s - I'll remove capacity stuff and not limit `buffer` in creation - just gonna add byte size comparison

Reply:
Yeah that's a better approach

Reply:
Updated!